build.assets/genproto.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+#
+# Builds, formats, lints and generates protos for teleport and teleport/api.
+set -eu
+
+main() {
+  cd "$(dirname "$0")"  # ./build-assets/
+  cd ../                # teleport root
+
+  buf build
+  buf lint
+  buf format -w
```

Comment:
Building teleterm's protos in the same manner as we build the rest of protos is a separate topic (last time I checked it's still not possible on M1 without building some tools from source).

However, would you consider adding some more changes to this PR so that those two commands run as part of the `buildbox-grpc-teleterm` target? The target already runs clang format + `buf generate` so I think it should be simple enough to run `buf lint` and `buf format` too (unless sharing lint & format config is a problem).

https://github.com/gravitational/teleport/blob/6e246ab5c9217737018e21b0c235d2df8ebcb911/Makefile#L967-L973

Reply:
Happy to follow up, but do you mind if I do it in a separate PR? I wanted to keep this one focused to api/ only.

We could do a bit more there, like using the same Buf version for both.

Reply:
Ah, sure thing!

Reply:
FYI: https://github.com/gravitational/teleport/compare/codingllama/protos-buf...codingllama/protos-teleterm?expand=1